standalone-examples/src/main/java/io/pravega/example/consolerw/ConsoleReader.java

```diff
@@ -30,67 +45,229 @@
 import io.pravega.client.stream.ReaderGroupConfig;
 import io.pravega.client.stream.ReinitializationRequiredException;
 import io.pravega.client.stream.ScalingPolicy;
-import io.pravega.client.stream.Sequence;
 import io.pravega.client.stream.StreamConfiguration;
 import io.pravega.client.stream.impl.JavaSerializer;
 
 /**
- * Reads from a Stream until interrupted.
+ * This class implements a simple console interface with the client for demonstration purposes. Specifically, this class
+ * has two main objectives: i) Reads from a configured {@link Stream} until interrupted; ii) It allows developers to
+ * have an easy first-time interaction with {@link StreamCut} API.
  */
 public class ConsoleReader {
     
-    private static final int READER_TIMEOUT_MS = 200000;
-    
-    public final String scope;
-    public final String streamName;
-    public final URI controllerURI;
+    private final String scope;
+    private final String streamName;
+    private final URI controllerURI;
+    private final MyReader reader;
+
+    private Map<Stream, StreamCut> streamCut;
+
+    private static final String[] MENU_TEXT = {
```

Comment:
Suggestion: The `org.apache.commons.cli.HelpFormatter` generates help for command line options. Could we use the same for these new options? 
It will also help eliminate the switch case statement below right?

Reply:
This could be an option, indeed. While this application just follows the pattern of existing ones in terms of CLI interactions, we can think in better ways to do this as a refactoring of CLI examples as a separate PR. This may include the suggestion from @fpj of using a single CLI console for all the examples.